lib/TestModule.py

```diff
@@ -0,0 +1,70 @@
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from cbor import cbor
+import pem
+import hexdump
+from typing import Optional, Tuple
+from aiocoap.error import NetworkError
+from aiocoap import *
+from logging import error, info
+import asyncio
+import logging
+import cbor
+import subprocess
+
+logging.basicConfig(level=logging.INFO)
+BASE_URI = 'coap://169.254.0.1/api/v1'
+
+requests = { # endpoint, method, input, output
+    '/admin/token_provision':{'method': Code.POST,'in': 'cbor','out': 'csr'},
+    '/admin/provision_complete':{'method': Code.POST,'in': 'csr','out': None}
+}
+
+class TestModule(object):
+    ROBOT_LIBRARY_VERSION = '1.0.0'
+    ROBOT_LIBRARY_SCOPE = 'GLOBAL'
+    
+    def convert_pem_to_cbor(self, input_file, output_file):
+
+        chain = pem.parse_file(input_file)
+        cert = []
+        for c in chain:
+            c = x509.load_pem_x509_certificate(
+                str(c).encode('utf-8'), default_backend())
+            der = c.public_bytes(serialization.Encoding.DER)
+            cert.append(bytes(der))
+        encoded = cbor.dumps({
+            'certs': cert
+        })
+
+        with open(output_file, 'wb') as out:
+            out.write(encoded)
+            out.close()
```

Comment:
There is no need to save output to a file, cbor should be returned to robotframework as object.

Reply:
applied